packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -219,6 +261,67 @@ - (UIImage *)scaleImage:(UIImage *)image by:(id)scaleParam {
   return image;
 }
 
+/**
+ * Creates a scaled version of the provided UIImage based on a specified scale factor. If the scale
+ * factor differs from the image's current scale by more than a small epsilon-delta (to account for
+ * minor floating-point inaccuracies), a new UIImage object is created with the specified scale.
+ * Otherwise, the original image is returned.
+ *
+ * @param image The UIImage to scale.
+ * @param scale The factor by which to scale the image.
+ * @return UIImage Returns the scaled UIImage.
+ */
+- (UIImage *)scaleImage:(UIImage *)image withScale:(CGFloat)scale {
+  if (fabs(scale - image.scale) > 1e-3) {
+    return [UIImage imageWithCGImage:[image CGImage]
+                               scale:scale
+                         orientation:(image.imageOrientation)];
+  }
+  return image;
+}
+
+/**
+ * Scales an input UIImage to a specified size. If the aspect ratio of the input image
+ * closely matches the target size, indicated by a small epsilon-delta, the image's scale
+ * property is updated instead of resizing the image. If the aspect ratios differ beyond this
+ * threshold, the method redraws the image at the target size.
+ *
+ * @param image The UIImage to scale.
+ * @param size The target CGSize to scale the image to.
+ * @return UIImage Returns the scaled UIImage.
+ */
+- (UIImage *)scaleImage:(UIImage *)image toSize:(CGSize)size {
+  if (fabs(((int)image.size.width * image.scale) - size.width) > 0 ||
+      fabs(((int)image.size.height * image.scale) - size.height) > 0) {
+    if (fabs(image.size.width / image.size.height - size.width / size.height) < 1e-2) {
```

Comment:
can you move this magic number as constant?

Reply:
removed the epsilon and replaced the functionality with a bit clearer code